tests/integration-tests/processors/processors-test.js

```diff
@@ -186,6 +190,98 @@ describe('Processors api', function() {
+                const createResponse = await requestSender.createProcessor(fileDownloadProcessor, validHeaders);
+                should(createResponse.statusCode).equal(201);
+                const processorId = createResponse.body.id;
+
+                const processor = Object.assign({}, fileDownloadProcessor);
+                processor.file_url = undefined;
+                processor.type = PROCESSOR_TYPE_RAW_JAVASCRIPT;
+                processor.javascript = 'module.exports = 1;';
+                const updateResponse = await requestSender.updateProcessor(processorId, processor, true);
+                should(updateResponse.statusCode).equal(200);
+                should(updateResponse.body.type).equal(processor.type);
+                should(updateResponse.body.javascript).equal(processor.javascript);
+                should(updateResponse.body.file_url).equal(undefined);
+            });
+            it('from raw_javascript processor to file_download', async function() {
+                const processor = generateRawJSProcessor('raw_js');
+                const createResponse = await requestSender.createProcessor(processor, validHeaders);
+                should(createResponse.statusCode).equal(201);
+                const processorId = createResponse.body.id;
+
+                const jsContent = 'module.exports = 25';
+                fileDownloadProcessor.file_url = baseURL + firstFileUri;
+                nock(baseURL).get(firstFileUri).reply(200, jsContent);
+                const updateResponse = await requestSender.updateProcessor(processorId, fileDownloadProcessor, true);
+                should(updateResponse.statusCode).equal(200);
+                should(updateResponse.body.type).equal(fileDownloadProcessor.type);
+                should(updateResponse.body.javascript).equal(jsContent);
+                should(updateResponse.body.name).equal(fileDownloadProcessor.name);
+            });
+            it('update a processor with dl = false', async function() {
+                nock(baseURL).get(firstFileUri).reply(200, 'module.exports = 5');
+                const createResponse = await requestSender.createProcessor(fileDownloadProcessor, validHeaders);
+                should(createResponse.statusCode).equal(201);
+                const processorId = createResponse.body.id;
+                const jsContent = createResponse.body.javascript;
+
+                const updatedProcessor = Object.assign({}, fileDownloadProcessor);
+                updatedProcessor.name = 'new name';
+                updatedProcessor.file_url = baseURL + secondFileUri;
+                nock(baseURL).get(secondFileUri).reply(200, 'const a = 5;');
+                const updateResponse = await requestSender.updateProcessor(processorId, updatedProcessor, false);
+                should(updateResponse.statusCode).equal(200);
+                should(updateResponse.body.name).equal(updatedProcessor.name);
```

Comment:
add another validation of `fileDownloadProcessor.description` and `fileDownloadProcessor.file_url`

Reply:
Correct me if I am wrong but I validated the `file_url` in line `#283`

Reply:
you included it perfectly, missed it